mono/mini/interp/transform.c

```diff
@@ -3539,17 +3569,9 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 				goto_if_nok (error, exit);
 			}
 
-			int mt = mint_type (m_class_get_byval_arg (klass));
-
-			interp_add_ins (td, (mt == MINT_TYPE_VT) ? MINT_LDOBJ_VT: MINT_LDOBJ);
-			td->last_ins->data [0] = get_data_item_index(td, klass);
+			interp_emit_ldobj (td, klass);
 
-			if (mt == MINT_TYPE_VT) {
-				size = mono_class_value_size (klass, NULL);
-				PUSH_VT (td, size);
-			}
 			td->ip += 5;
```

Comment:
do `MINT_LDOBJ_VT` and `MINT_LDIND_I1_CHECK`/`MINT_LDIND_I2_CHECK`/etc. have the same instruction length?

Reply:
I don't see why it matters here. This td->ip is for the IL stream not for the MINT code

Reply:
ah, sorry